src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.Analyzer.cs

```diff
@@ -25,6 +25,9 @@ public static AnalyzerResult Analyze(CSharpSelectionResult selectionResult, bool
         protected override bool TreatOutAsRef
             => false;
 
+        protected override bool IsInPrimaryConstructorBaseType()
+            => this.SelectionResult.GetContainingScopeOf<PrimaryConstructorBaseTypeSyntax>() != null;
```

Comment:
the majority of the pr is just updating all the places in EM that special case certain locations to now special case PrimaryConstructorBaseTypeSyntax as well.